go/mysql/collations/charset/eightbit/8bit.go

```diff
@@ -81,3 +81,17 @@ func (Charset_8bit) Length(src []byte) int {
 func (Charset_8bit) MaxWidth() int {
 	return 1
 }
+
+func (Charset_8bit) Slice(src []byte, from, to int) []byte {
+	if from >= len(src) {
+		return nil
+	}
+	if to > len(src) {
+		to = len(src)
+	}
+	return src[from:to]
+}
```

Comment:
Added a number of these optimized / simplified implementations where possible as well (we check for this interface when slicing in the helper).